domain/topn_slow_query.go

```diff
@@ -59,28 +62,79 @@ func (h *slowQueryHeap) Refresh(now time.Time, recent time.Duration) {
 	heap.Init(h)
 }
 
+func (h *slowQueryHeap) Query(count int) []*SlowQueryInfo {
+	// The sorted array still maintains the heap property.
+	sort.Sort(h)
+
+	// The result shoud be in decrease order.
+	return takeLastN(h.data, count)
+}
+
+type slowQueryQueue struct {
+	data []*SlowQueryInfo
+	size int
+}
+
+func (q *slowQueryQueue) Enqueue(info *SlowQueryInfo) {
+	if len(q.data) < q.size {
+		q.data = append(q.data, info)
+		return
+	}
+
+	q.data = append(q.data, info)[1:]
```

Comment:
If you just use slice here, the q.data will keep the continuous memory, it may cause oom. What about using a list?

Reply:
Do you do a large amount insert test?

Reply:
@coocood suggest me to use this.
The code is the simplest queue implementation, although it double the memory usage.

Reply:
The queue size will be 500.
We just keep the most recent queries in the queue.